### PR TITLE
fix: improvements to the wizard in dialog

### LINKED ISSUE #4342: Wizard / dialog refactors needed to match designs
Currently the wizard and dialog may need some refactoring in order to cooperate together according to the Fiori specs.

1. Scroll behavior - Wizard progress bar should stay fixed.  Dialog should not use dialog header and footer. Only the content should scroll.  The content height calculation is also incorrect for dialogs, as the current implementation assumes the entire wizard will occupy `100vh`.

2. Incorrect font size for dialog title

3. Footer should be a floating bar

4. In Safari (not pictured) top left and top right dialog corners are missing border radius. Does not happen with other dialogs so I am currently unsure of the fix for this

UI5:
Screen Shot 2021-01-14 at 2 37 13 PM

fd-ngx
Screen Shot 2021-01-14 at 2 37 09 PM

2d

### PR DESCRIPTION
**Pull request by mikerodonnell89**
fixes #4342 

Adds many improvements to correct the wizard when used in a dialog, such as correcting font sizes, using floating footer, putting scrollbars on the correct elements etc.  Also contains improvements to the logic around fitting wizard content to the container.

Before:
Screen Shot 2021-02-04 at 10 39 51 AM

After:
Screen Shot 2021-02-04 at 10 40 16 AM

### CODE REVIEW

File: apps/docs/src/app/core/component-docs/wizard/examples/wizard-dialog-example.component.html [hunk 1]
```diff
@@ -219,52 +216,15 @@ <h4 class="fd-title fd-title--h4">4. Payment Information</h4>
                     </ul>
                 </fd-wizard-navigation>
             </fd-wizard>
+            <div fd-bar [barDesign]="'floating-footer'">
```
Comment (Betrozov) [inline apps/docs/src/app/core/component-docs/wizard/examples/wizard-dialog-example.component.html:219]: unnecessary binding
`[barDesign]="'floating-footer'"` -> `barDesign="floating-footer"`

File: apps/docs/src/app/core/component-docs/wizard/examples/wizard-mobile-example.component.ts [hunk 2]
```diff
@@ -26,6 +26,16 @@ export class WizardMobileExampleComponent {
...
     example2step3status: WizardStepStatus = 'upcoming';
 
+    statusChanged(exampleNumber: number, stepNumber: number, event: any): void {
+        if (event === 'current') {
+            if (exampleNumber === 1) {
+                this.example1goToStep(stepNumber);
+            } else if (exampleNumber === 2) {
+                this.example2goToStep(stepNumber);
+            }
+        }
```
Comment (Betrozov) [inline apps/docs/src/app/core/component-docs/wizard/examples/wizard-mobile-example.component.ts:36]: please decrease nesting
```suggestion
        if (event !== 'current') {
            return;
        }

        if (exampleNumber === 1) {
            this.example1goToStep(stepNumber);
        } else if (exampleNumber === 2) {
            this.example2goToStep(stepNumber);
        }
```

File: libs/core/src/lib/wizard/wizard.component.ts [hunk 3]
```diff
@@ -207,8 +225,15 @@ export class WizardComponent implements AfterViewInit, OnDestroy {
...
-                'calc(100vh - ' + combinedHeight + 'px)';
+            if (
+                wizard.querySelector('.' + WIZARD_TALL_CONTENT_CLASS).offsetHeight <
+                wizard.querySelector('.' + WIZARD_CONTAINER_WRAPPER_CLASS).offsetHeight
+            ) {
+                wizard.querySelector('.' + WIZARD_TALL_CONTENT_CLASS).style.height =
+                    'calc(100vh - ' + combinedHeight + 'px)';
+            } else {
+                wizard.querySelector('.' + WIZARD_TALL_CONTENT_CLASS).style.paddingBottom = '24px';
+            }
```
Comment (Betrozov) [inline libs/core/src/lib/wizard/wizard.component.ts:236]:
```suggestion
            const wizardTallContent = wizard.querySelector(`.${WIZARD_TALL_CONTENT_CLASS}`);
            const wizardContainerWrapper = wizard.querySelector(`.${WIZARD_CONTAINER_WRAPPER_CLASS}`);

            if (wizardTallContent.offsetHeight < wizardContainerWrapper.offsetHeight) {
                wizardTallContent.style.height = `calc(100vh - ${combinedHeight}px)`;
            } else {
                wizardTallContent.style.paddingBottom = '24px';
            }
```

**Review by Betrozov — CHANGES_REQUESTED** [review]

File: apps/docs/src/app/core/component-docs/wizard/examples/wizard-branching-example.component.ts [hunk 4]
```diff
@@ -40,6 +40,12 @@ export class WizardBranchingExampleComponent implements OnInit {
         this.oldPayment = this.paymentSelection;
     }
 
+    statusChanged(stepNumber: number, event: any): void {
+        if (event === 'current') {
+            this.goToStep(stepNumber);
+        }
+    }
+
```
Comment (JKMarkowski) [inline apps/docs/src/app/core/component-docs/wizard/examples/wizard-branching-example.component.ts:48]: Can this `event` have more specific type?

File: apps/docs/src/app/core/component-docs/wizard/examples/wizard-customizable-example.component.ts [hunk 5]
```diff
@@ -28,6 +28,12 @@ export class WizardCustomizableExampleComponent {
     step2status: WizardStepStatus = 'upcoming';
     step3status: WizardStepStatus = 'upcoming';
 
+    statusChanged(stepNumber: number, event: any): void {
+        if (event === 'current') {
+            this.goToStep(stepNumber);
+        }
+    }
+
```
Comment (JKMarkowski) [inline apps/docs/src/app/core/component-docs/wizard/examples/wizard-customizable-example.component.ts:36]: Can this event have more specific type?

**Review by JKMarkowski — COMMENTED** [review]
Hi @mikerodonnell89 check my comment regarding typing, also let's propagate it to any other example. Rest LGTM

**Review by Betrozov — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Review by JKMarkowski — APPROVED** [review]
LGTM

**Pull request merged**